recipes/views.py

```diff
@@ -1,149 +1,156 @@
+#         'ingredients__ingredient').annotate(
+#             total=Count('ingredients')).order_by('-total')[:5]
+#
+#     context = {
+#         "recipes": recipes,
+#         "most_used_ingredients": most_used_ingredients
+#     }
+#
+#     return render(request, "index.html", context)
+#
+#
+# def index(request):
+#     all_recipes = Recipe.objects.all().order_by('-created_time')
+#     return main_page_view(request, all_recipes)
+#
+#
+# def search(request):
+#     terms = [x.strip() for x in request.GET.get("q").split(",")]
+#     ids = set()
+#     for term in terms:
+#         for j in Ingredient.objects.filter(ingredient__contains=term):
+#             ids.add(j.id)
+#     result_recipes = Recipe.objects.filter(
+#         ingredients__in=list(ids)).order_by("-created_time")
+#     for term in terms:
+#         tmp = Recipe.objects.filter(
+#             Q(title__contains=term)
+#             | Q(description__contains=term)).order_by('-created_time')
+#         result_recipes = result_recipes | tmp
+#
+#     all_recipes = result_recipes.distinct()
+#     return main_page_view(request, all_recipes)
+#
+#
+# def ingredient(request, ingredient_value):
+#     recipes = Recipe.objects.filter(ingredients__ingredient=ingredient_value)
+#     return main_page_view(request, recipes)
+#
+#
+# def recipe_detail(request, pk):
+#     recipe = Recipe.objects.get(id=pk)
+#     context = {
+#         "recipe": recipe,
+#     }
+#
+#     if request.user.is_authenticated:
+#         is_liked = Like.objects.filter(
+#             user=request.user, recipe=recipe).count()
+#
+#         is_rated = Rate.objects.filter(
+#             user=request.user, recipe=recipe).count()
+#
+#         rate_point = 0
+#         if is_rated:
+#             rate_point = Rate.objects.get(
+#                 user=request.user, recipe=recipe).score
+#
+#         extra_context = {
+#             "is_liked": is_liked,
+#             "is_rated": is_rated,
+#             "rate_point": rate_point,
+#         }
+#
+#         context = {**context, **extra_context}
+#
+#     return render(request, 'recipe_detail.html', context)
+#
+#
+# @login_required
+# def like_recipe(request, pk):
+#     recipe = Recipe.objects.get(pk=pk)
+#
+#     try:
+#         like = Like.objects.get(user=request.user, recipe=recipe)
+#         like.delete()
+#         return redirect(recipe_detail, recipe.id)
+#
+#     except Like.DoesNotExist:
+#         Like.objects.create(user=request.user, recipe=recipe)
+#         return redirect(recipe_detail, recipe.id)
+#
+#
+# @login_required
+# def rate_recipe(request, pk):
+#     recipe = Recipe.objects.get(id=pk)
+#     rate_score = request.POST.get('score')
+#     rate, created = Rate.objects.update_or_create(
+#         user=request.user, recipe=recipe,
+#         defaults={'score': rate_score},
+#     )
+#     return redirect(recipe_detail, recipe.id)
+
+
+class ListCreateIngredientView(ListCreateAPIView):
```

Comment:
There is no permission control for this view. So, someone can create tons of ingredients to slow down you system without registering. It is not absolute solution but we can give permission to only our users.